modules/services/window-managers/xmonad.nix

```
@@ -75,27 +75,57 @@ in {
           by Home Manager.
         '';
       };
+
+      libFiles = mkOption {
+        type = types.attrs;
+        default = { };
+        example = literalExample ''
+          {
+            "Tools.hs" = pkgs.writeText "Tools.hs" '''
+               module Tools
+               where
+
+               screenshot = "scrot"
+             ''';
+          }
+        '';
+        description = ''
+          Additional files that will be saved in
+          <filename>~/.xmonad/lib/</filename>. The keys are the file names
+          the value is a path to the contents of the files.
```

```suggestion
          <filename>~/.xmonad/lib/</filename>. The keys are the file names while
          the values are paths to the contents of the files.
```

What do you think of this?